lib/article_json/export/facebook_instant_article/elements/embed.rb

```diff
@@ -3,7 +3,40 @@ module Export
     module FacebookInstantArticle
       module Elements
         class Embed < Base
-          include ArticleJSON::Export::Common::HTML::Elements::Embed
+          include ArticleJSON::Export::Common::HTML::Elements::Shared::Caption
+
+          # Generate the embedded element node
+          # @return [Nokogiri::XML::NodeSet]
+          def export
+            create_element(:figure, class: 'op-interactive') do |figure|
+              figure.add_child(embed_node)
+              if @element.caption&.any?
+                figure.add_child(caption_node(:figcaption))
+              end
+            end
+          end
+
+          private
+
+          # Type specific object that should be embedded
+          # @return [Nokogiri::XML::Element]
+          def embed_node
+            if %i(facebook_video tweet).include? @element.embed_type.to_sym
+              default_node
```

Comment:
I would expect the default to be in the `else` branch, and not for two specific cases...

Reply:
I changed the method name to 'iframe_node'.

Reply:
thanks!